new header with collapsing menu

rnevet opened the pull request:
A new header (better bootstrap usage)
Restored the default refresh rate to 30 as it was, no need to waste bandwidth. :)

Commit:
new html header with collapsing menu

Commit:
restore default refreshrate to 30 seconds

Commit:
added a version parameter to the javascript to force reload

Commit:
added missing scheme in toastr urls

Commit:
rename loadSave to loadSettings

Commit:
Added Bitcoin Display Unit to Settings.

Commit:
Added OutputCurrencyDisplayMode to Settings.

Commit:
Added Effective Rate Calculation to Settings.

ckcollab commented:
30 default makes sense

rnevet committed:
Fix indentation issues
Replace printFloat with prettyfloat for Fiat currencies (doesn't display .00)

rnevet commented:
All comments should be fixed.

rnevet commented:
Also replace printFloat with prettyFloat to prevent 0.00 USD value. (prettyFloat will always make sure to show you at least a precision of 2 after first non zero decimal)

printFloat:
0.41 EUR / Year
0.03 EUR / Month
0.00 EUR / Day

prettyFloat
0.41 EUR / Year
0.032 EUR / Month
0.00105 EUR / Day

ckcollab commented on www/lendingbot.html:
```diff
@@ -34,29 +34,35 @@
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/toastr.min.css" rel="stylesheet">
 
-    <script src="lendingbot.js" ></script>
+    <script src="lendingbot.js?version=1" ></script>
```
Is there a nicer way to do this? So we don't have to manually change this version?

rnevet replied:
Maybe, lets keep it for this PR. Already too many changes.

ckcollab commented on www/lendingbot.html:
```diff
@@ -65,20 +71,36 @@
-                    <span id="updated">Not updated</span>
-                    <i class="glyphicon glyphicon-cog" data-toggle="modal" data-target="#settings_modal" style="color: orangered; cursor: pointer;"></i>
+
+                <!-- Brand and toggle get grouped for better mobile display -->
```
Indented too far

ckcollab commented on www/lendingbot.js:
```diff
@@ -370,6 +376,30 @@ function doSave() {
     localStorage.setItem('timespanNames', JSON.stringify(timespanNames))
 
+    // Bitcoin Display Unit
+    localStorage.displayUnitText = $('input[name="btcDisplayUnit"]:checked').val();
```
I don't think you're using local storage right here...?

rnevet replied:
That looks correct, why?

ckcollab replied:
What are you trying to do? That'd be like going:

`jQuery.displayUnitText = something`

Just do

`displayUnitText = something`

ckcollab replied:
Use get item and set item

ckcollab replied:
Wow, I didn't realize this worked!

However, just because we don't have `protected` or `private` here, we shouldn't really skip past the getters/setters... http://stackoverflow.com/questions/12632463/is-localstorage-getitemitem-better-than-localstorage-item-or-localstoragei

> *As Christoph points out in his answer, the functional interface is the only way to reliably store and retrieve values under keys equal to the predefined properties of localStorage*

We're setting ourselves up to get shot in the foot, directly accessing these parameters and muddying up `localStorage`. Also, it's inconsistent doing it two ways.

`set/getItem` is how I'd recommend we do it.

ckcollab replied:
Also just noticed, I believe this can be refactored:

```
    validOutputCurrencyDisplayModes.forEach(function(mode) {
        if(mode == localStorage.outputCurrencyDisplayModeText) {
            outputCurrencyDisplayMode = mode;
        }
    })
```

to

```
    if(validOutputCurrencyDisplayModes.indexOf(mode) !== -1) {
        outputCurrencyDisplayMode = mode;
    }
```

rnevet replied:
Correct, this is your branch, feel free. ;)

rnevet replied:
I also agree about setters and getters, however this is already in use in other places in this file. I actually just reused the method.

ckcollab replied:
Cool, that's fine then, I just tried refactoring that validation via the online editor, apologies if I made a small mistake or something -- should work though.

rnevet commented:
This line saves the checked value to local storage.

ckcollab commented:
> prettyFloat
0.41 EUR / Year
0.032 EUR / Month
0.00105 EUR / Day

Nice, handy 👍

ckcollab committed:
refactor validating... will still behave weirdly if somehow manages to get bad data set (sets localStorage without checking if it's valid)

rnevet committed:
Fixed indent issue

rnevet committed:
Restored btcDisplayUnitsModes saving method

As btcDisplayUnitsModes isn't a list of strings a simple indexOf won't match the mode.

rnevet commented:
@ckcollab Think we can merge this into patch-1 so we can test it, with both of us making changes on the web I'm not sure which branch contains everything.
